source/codegen/service_helpers.py

```diff
@@ -422,6 +422,24 @@ class CrossDriverSessionDependency(NamedTuple):
     local_name: str
 
 
+def get_driver_shared_resource_repository_ptr_deps(
+    driver_config: dict,
+) -> List[CrossDriverSessionDependency]:
+    resource_repository_deps = [
+        _create_cross_driver_session_dependency(resource_handle_type)
+        for resource_handle_type in get_resource_handle_types(driver_config)
+    ]
+
+    resource_repository_deps[0] = CrossDriverSessionDependency(
+        resource_repository_deps[0].resource_handle_type,
+        "ResourceRepositorySharedPtr",
+        resource_repository_deps[0].resource_repository_type,
+        "session_repository_",
```

Comment:
note: I haven't changed this to `resource_repository_` as this will essentially cause every function from every driver to have single line change. That's too much noise for this PR. I can send follow up PR once this is done to update field name for default repository.

Reply:
Sadly you get noise either way :(